lib/private/Setup.php

```diff
@@ -560,6 +560,14 @@ public static function updateHtaccess() {
 		}
 
 		if ($content !== '') {
+			// Never write file back if disk space should be too low
+			if (function_exists('disk_free_space')) {
+				$df = disk_free_space(\OC::$SERVERROOT);
+				$size = strlen($content) + 10240;
+				if ($df !== false && $df < (float)$size) {
+					throw new \Exception(\OC::$SERVERROOT . " does not have enough space for writing the htaccess file! Not writing it back!");
+				}
+			}
 			//suppress errors in case we don't have permissions for it
 			return (bool)@file_put_contents($setupHelper->pathToHtaccess(), $htaccessContent . $content . "\n");
```

Comment:
You can also check if all bytes have been written. Should be something like this:
```suggestion
			$fullContent = $htaccessContent . $content . "\n";
			return @file_put_contents($setupHelper->pathToHtaccess(), $$fullContent) === strlen($fullContent);
```

Reply:
This is not going to help here. The prpblem is that file_pu_content truncates the file if there is not enough space which should never happen as the file is very important to work...

Reply:
We had the same prpblem with config.php before and fixed in in the same way.

Reply:
It's another security in case something happen during the copy.
The file will already be erased, but at least you'll have the info